libs/contrib/Data/Fin/Extra.idr

```diff
@@ -97,3 +123,49 @@ divMod {ok=_} (S n) (S d) =
         Right (r' ** eq') => Fraction {ok=ok} (S n) (S d) q (FS r') $
             rewrite sym $ plusSuccRightSucc (q * S d) (finToNat r') in
                 cong S $ trans (sym $ cong (plus (q * S d)) eq') eq
+
+-------------------------------------------------
+--- Splitting operations and their properties ---
+-------------------------------------------------
+
+export
+splitSumFin : {a : Nat} -> Fin (a + b) -> Either (Fin a) (Fin b)
+splitSumFin {a=Z}   x      = Right x
+splitSumFin {a=S k} FZ     = Left FZ
+splitSumFin {a=S k} (FS x) = bimap FS id $ splitSumFin x
```

Comment:
`bimap f id` is `mapFst f`

Reply:
I believed that this function exists, but I failed to find it, shame on me